### PR TITLE
Add danger

### LINKED ISSUE #3: Add a changelog
Should we go for semantic commits so we don't have to manually maintain it?

Should we use semantic release and just publish from CI?

### LINKED ISSUE #7: Add Danger

### PR DESCRIPTION
**Pull request by orta**
Adds:

- Danger - fixes #7 
- A rule to check that a PR has a body
- A rule to check that changes to `rules/*` includes a CHANGELOG entry
- A CHANGELOG - fixes #3 

screen shot 2017-11-13 at 8 10 22 am

### CODE REVIEW

**Comment by orta** [comment]
Hrnm, interesting - looks like one of Danger's dependencies is a `node 6 > dep`. So I've made danger get added at runtime.

**Comment by SimenB** [comment]
Woo!

Still failing CI because of node 6.

Would using build stages allow us to only run danger on one env (e.g. node 8)? https://docs.travis-ci.com/user/build-stages

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+// Please add your own contribution below inside the Master section // These
+docs are aimed at users rather than danger developers, so please limit technical
```
Comment (SimenB) [inline CHANGELOG.md]: "danger" should probably not be here

Comment (orta) [reply CHANGELOG.md]: Agree 👍

**Comment by SimenB** [comment]
I've added a separate changelog, causing a conflict.

(I still want to use conventional commits and just generate it, but now it's there, at least)

**Comment by SimenB** [comment]
I've now deleted the changelog as semantic-release is setup. I'd still love to have the test report from jest and lint report from eslint inline. Danger supports that, right?

Not sure how to deal with node 4. Keen on dropping it, it messes up for semantic release as well :P

**Comment by orta** [comment]
Hah - sorry for letting this drop for so long, still mid-way on a big deadline, but I've switched it up to have danger only run on 8 and to remove the CHANGELOG rule. Right now it only has one simple "all PRs must have a description" rule. Can expand later 👍

**Comment by orta** [comment]
OK - that's that 👍

File: .travis.yml [hunk 2]
```diff
@@ -22,6 +22,13 @@ jobs:
     after_success:
     - yarn add -D semantic-release@11.0.1
     - yarn semantic-release
+  - stage: danger
+    node_js: 8
+    os: linux
+    script: 
+      - yarn add -D danger@^2
```
Comment (SimenB) [inline .travis.yml]: `npx danger@^2`?

File: dangerfile.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+const { danger, fail } = require('danger');
+
+// Ensure that people include a description on their PRss
```
Comment (SimenB) [inline dangerfile.js]: `PRss` typo?

File: .travis.yml (hunk 2 above)
Comment (orta) [reply .travis.yml]: @cpojer where is my `yarn exec danger` ;)

Comment (orta) [reply .travis.yml]: Agree though @SimenB

Comment (SimenB) [reply .travis.yml]: https://github.com/yarnpkg/yarn/issues/3937

**Comment by orta** [comment]
Nice updated 👍

**Comment by SimenB** [comment]
```sh-session
$ npx danger@^2
npx: installed 110 in 4.601s
child process exited with code 0
Request failed [404]: https://api.github.com/repos/jest-community/eslint-plugin-jest/statuses/f66f69562d184115afb2865338910b717be15d1f
Response: {
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3/repos/statuses/#create-a-status"
}
No issues or messages were sent. Removing any existing messages.
```

Are we supposed to get the 404?

**Comment by SimenB** [comment]
We've added `ignore-engines`, so you should be able to add `danger` as a dep now.

**Comment by orta** [comment]
Lols,

**Comment by macklinu** [comment]
@orta let me know if I can help out with this one! 👋

**Comment by orta** [comment]
Ah yeah, the CI fail is nothing to do with danger

**Comment by orta** [comment]
Rebased 👍

**Comment by orta** [comment]
The trick was the ordering of the tasks in the travis.yml 👍

**Comment by SimenB** [comment]
Does this matter?

>You may have updated from Danger 2.x -> 3.x without updating from `danger` to `danger ci`.

https://travis-ci.org/jest-community/eslint-plugin-jest/jobs/340160669#L471

**Pull request merged**